Foundation/AffineTransform.swift

```diff
@@ -9,8 +9,10 @@
 
 #if os(macOS) || os(iOS)
     import Darwin
-#elseif os(Linux) || CYGWIN
+#elseif os(Linux) || os(Android) || CYGWIN
     import Glibc
+#elseif os(Windows)
+    import MSVCRT
```

Comment:
this seems to be growing yet again, perhaps we should find a way to do this in one spot via `@_exported import` for all of these in just one file.

Reply:
just tested it out; it seems like we can delete all of them and just have them all in NSSwiftRuntime.swift which seems much better to have one spot of failure and one spot for changes in the future (also worth commentary there to that extent informing future contributors this is the spot to import platform libc type things)

Reply:
Interestingly it seems that the `import Darwin` can be removed from `NSSwiftRuntime.swift` as well. I think the Xcode project might be pulling in a header file that ends up including all of the relevant C header files. However on Linux it still requires the `import Glibc` in `NSSwiftRuntime.swift`. Still a good simplification though.

Reply:
Updated #1842 for this.  I'm trying to split up the changes into something more digestible so that we can look at the specifics of the local changes (e.g. just NSLock) rather than everything at once.  Unfortunately, I have a bunch of changes that are foundational (ha ha) to the rest of the changes.